### PR TITLE
End to end test

### PR DESCRIPTION
**Pull request by dcolligan**
change default sink values from empty strings to none

add end to end test

adds test annotations

### CODE REVIEW

**Comment by jeromekelleher** [comment]
This looks good, but we need to change the way the server CLI is invoked. Probably the easiest thing to do is to write a configuration to a temp file, and pass this path to the server as a '-f' argument. This should make it easy to control the configuration of the server.

What are the start-up/shutdown times like here?

**Comment by dcolligan** [comment]
```
$ nosetests tests/end_to_end/
...
Ran 1 test in 1.384s
```

**Comment by dcolligan** [comment]
The time is mainly dependent on the utils.Repeat.sleepSeconds constant.  If we bump it down to 0.1 the execution time drops to 0.658s.  I'll make that change.

**Comment by jeromekelleher** [comment]
OK, that's good, we can run these every time then. Excellent.

**Comment by dcolligan** [comment]
@jeromekelleher why can't we just use the TestConfig for the server (as I do here)?

File: ga4gh/cli.py [hunk 1]
```diff
@@ -246,6 +246,9 @@ def addGlobalOptions(parser):
     parser.add_argument(
         "--config-file", "-f", type=str, default=None,
         help="The configuration file to use")
+    parser.add_argument(
+        "--dont-use-reloader", default=False, action="store_true",
```
Comment (jeromekelleher) [inline ga4gh/cli.py]: Maybe `--disable-reloader`?

**Comment by jeromekelleher** [comment]
Yes, that's a better option actually. Can we put this on a different port please? It's a bit annoying to have to stop my dev server each time I want to run the unittests. Something large and random seems good.

Otherwise, +100. This is really tidy and runs really quickly and smoothly.

**Comment by dcolligan** [comment]
Alright, 8001 it is...

ready to merge

**Pull request merged**